recipe/build.sh

```diff
@@ -82,3 +86,15 @@ ctest -VV
 # Leave this test where it is. ATM, conda-build deletes host prefixes by the time it runs the
 # package tests which makes investigating problems very tricky. Pinging @msarahan about that.
 ncdump/ncdump -h http://geoport-dev.whoi.edu/thredds/dodsC/estofs/atlantic || exit $?
+
+if [[ ${c_compiler} != "toolchain_c" ]]; then
+    # Fix build paths in cmake artifacts
+    for fname in `ls ${PREFIX}/lib/cmake/netCDF/*`; do
+        sed -i.bak "s#${BUILD_PREFIX}#\$ENV\{BUILD_PREFIX\}#g" ${fname}
```

Comment:
Ack. Looks like @ocefpaf missed this comment. :(

My motivation here is that cmake in downstream builds appears to look for stuff in the sysroot for the packaged anaconda compilers, which is in the build prefix. 

As far as I can tell, the host prefix stuff is taken care of by conda build on its own by detection the host prefix path and replacing it.

Reply:
The comment was done on my phone and marked as pending which I didn't notice until I booted up a computer, so oit was my fault.

The build prefix is different from the host prefix, and you *definitely* want the host prefix in this case, trust me on that.

Reply:
To be clear, the build prefix only comes with CDT packages and compiler runtimes otherwise it should be populated with stuff that may not even run on the host architecture (think cross-compilation).

Other than those special cases, libraries that link to the host package (i.e. the one you are building) should always be in `requirements/host` and should always self-refer to any of their sub components via paths prefixed with the host prefix.

Reply:
Right I totally believe you. I am def a n00b on this stuff. Can you help me understand the downstream error then?

Here is the error from the downstream build (https://circleci.com/gh/conda-forge/netcdf-fortran-feedstock/52?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link):

```make[2]: *** No rule to make target `/home/conda/feedstock_root/build_artifacts/libnetcdf_1542306222682/_build_env/x86_64-conda_cos6-linux-gnu/sysroot/usr/lib/librt.so', needed by `nf_test/tst_f77_v2'```

For whatever reason, libnetcdf has an explicit reference to an `so` that is in the sysroot. When the downstream build happens, this `so` will be in the build prefix of the downstream build, not the host prefix, since the compilers are always listed in the build, not host section. This is why I used the build prefix above. 

How do I square this all with what you are saying?

Reply:
> I also checked the libgcc-ng and libstdcxx-ng packages for the gcc7 linux compilers and they don't have the so

The so is part of either the compiler package itself or maybe binutils I cannot remember (it is compiled as part of glibc). Regardless, CDT libraries and glibc libraries are only *proxies* for the system provided libraries (this is so that, e.g. users of Anaconda Distribution / conda-forge on Clear Linux still take full advantage of the *massive* speed-ups in recent glibc versions => they use e.g. AVX512 for string operations when running on compatible hardware), they should never be installed on end-user machines, the distro-provided versions are always to be used instead.

This is some CMake inflexibility. I suggest you sed it out of the final files and move on!

Reply:
As an alternative to using sed here you could take advantage of the fact that librt is redundant now-a-days (has been for years) and the functionality from it is now all part of glibc's libc itself. Luckily CMake hasn't decided (yet!) to do any special handling of libc.

Reply:
Thanks for the reply! 

So your suggestion is to completely remove any references to anything in the build prefix? (I didn't understand the phrase "sed it out." :( )

I'm happy to make that PR!

Reply:
It seems librt was deprecated in glibc 2.17 so that's too recent for us.

sed it out means to get rid of this librt dependency via sed (using either the d or s options). It shouldn't appear at all since it's a transitive dependency only that's satisfied by (and an implementation detail of) libnetcdf.